src/freenet/client/filter/GenericReadFilterCallback.java

```diff
@@ -274,7 +260,49 @@ public String processURI(String u, String overrideType, boolean forBaseHref, boo
 			throw new CommentException(l10n("protocolNotEscaped", "protocol", uri.getScheme()));
 		}
 	}
-	
+
+	@Override
+	public String processURI(String u, String overrideType, String forceHostPort)
+			throws CommentException {
+		URI uri;
+		String filtered;
+		try {
+			filtered = processURI(makeURIAbsolute(u), overrideType, true, false);
+			uri = URIPreEncoder.encodeURI(filtered).normalize();
+		} catch (URISyntaxException e1) {
+			if(logMINOR) Logger.minor(this, "Failed to parse URI: "+e1);
+			throw new CommentException(l10n("couldNotParseURIWithError", "error", e1.getMessage()));
+		}
+		if (uri.getHost() == null) {
```

Comment:
I’m not sure I understand this block… in what cases does a URI not have a host? Do basically all freenet URIs not have a host?

Reply:
Okay, after briefly checking `processURI` I think it’s clear to me now. :)